test/msgpax_test.exs

```diff
@@ -209,5 +214,8 @@ defmodule MsgpaxTest do
     end
 
     assert Msgpax.pack!(%UserWithAge{name: "Luke", age: 9}) == Msgpax.pack!(%{name: "Luke"})
+
+    assert Msgpax.pack!(%UserDerivingStruct{name: "Juri"}) ==
+           Msgpax.pack!(%{"__struct__" => UserDerivingStruct, name: "Juri"})
```

Comment:
Maybe bind `Msgpax.pack!(%{"__struct__" => UserDerivingStruct, name: "Juri"})` to `expected`?